stock_location_tank/models/stock_picking.py

```diff
@@ -0,0 +1,165 @@
+        for rec in self:
+            if rec.location_dest_id.is_tank:
+                rec.env.context = dict(rec.env.context)
+                rec.env.context["allow_duplicate"] = True
+        return super(StockPicking, self).button_validate()
+
+    def _action_done(self):
+        res = super(StockPicking, self)._action_done()
+        for rec in self:
+            # Juntar els move_line_ids que tinguin el mateix product_id i location_id
+            moves_lines = {}
+            for line in rec.move_line_ids_without_package:
+                key = (line.product_id.id, line.location_dest_id.id)
+                if key in moves_lines:
+                    moves_lines[key] += line.qty_done
+                else:
+                    moves_lines[key] = line.qty_done
+            for (prod, loc), qty_done in moves_lines.items():
+                product = rec.env["product.product"].browse(prod)
+                location_dest = rec.env["stock.location"].browse(loc)
+                if not location_dest.is_tank:
+                    continue
+                if not product in location_dest.product_ids:
+                    raise UserError(
+                        _("Product %s not allowed in tank location %s")
+                        % (product.name, location_dest.name)
+                    )
+                if product.uom_id != location_dest.product_uom_id:
+                    raise UserError(
+                        _(
+                            "Product %s cannot have different Unit of Measure than tank location %s"
+                        )
+                        % (product.name, location_dest.name)
+                    )
+                warehouse = rec.picking_type_id.warehouse_id
+                mrp_operation_type = rec.env["stock.picking.type"].search(
+                    [
+                        ("warehouse_id", "=", warehouse.id),
+                        ("code", "=", "mrp_operation"),
+                        ("tank_filling", "=", True),
+                    ]
+                )
+                if not mrp_operation_type:
+                    raise UserError(
+                        _("Not found mrp_operation picking type for tank location %s")
+                        % location_dest.name
+                    )
+                if len(mrp_operation_type) > 1:
+                    raise UserError(
+                        _(
+                            "More than one mrp_operation picking type for tank location %s"
+                        )
+                        % location_dest.name
+                    )
+                if not mrp_operation_type.default_location_src_id:
+                    raise UserError(
+                        _(
+                            "Not found default location source for mrp_operation picking type %s"
+                        )
+                        % mrp_operation_type.name
+                    )
+                component_quant = rec.env["stock.quant"].search(
```

Comment:
Will it work if the user has more than one company selected? Check if something similar could be happenning in other `search` in all the code.